server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -109,7 +110,9 @@ private static Bootstrap initPhase1() {
         final PrintStream out = getStdout();
         final PrintStream err = getStderr();
         final ServerArgs args;
-        final boolean useEntitlements = Boolean.parseBoolean(System.getProperty("es.entitlements.enabled"));
+        final boolean entitlementsExplicitlyEnabled = Booleans.parseBoolean(System.getProperty("es.entitlements.enabled", "false"));
+        // java 24+ only supports entitlements, but it may be enabled on earlier versions explicitly
+        final boolean useEntitlements = Runtime.version().feature() >= 24 || entitlementsExplicitlyEnabled;
```

Comment:
if you want, there is `RuntimeVersionFeature.isSecurityManagerAvailable()` (created so we can easily track the places where we make this check)

Reply:
Done in [575a198](https://github.com/elastic/elasticsearch/pull/119885/commits/575a1982bee609a5a53b7a817bc7a1ce1a5aedaa)